### ISSUE TITLE
flutter install command fails (iOS app)

### ISSUE THREAD

**Issue by ketanm** [flutter/flutter#42543]
Hi,

I recently upgraded Flutter to the latest version (`v1.9.1+hotfix.4`) and command that I was using earlier to install apps (`flutter install path_to_release.app`) started failing constantly with the following error:
```
Installing com.companyName.AppName to LUM-IPAD-KM...
Install failed

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      InstallCommand.runCommand (package:flutter_tools/src/commands/install.dart:44:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:490:18)
#3      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#4      _rootRunUnary (dart:async/zone.dart:1132:38)
#5      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#6      _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#7      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#8      Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#9      Future._completeWithValue (dart:async/future_impl.dart:522:5)
#10     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
#11     _rootRun (dart:async/zone.dart:1124:13)
#12     _CustomZone.run (dart:async/zone.dart:1021:19)
#13     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#14     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#15     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#17     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#18     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```

Switching back to the previous working state by doing `flutter version  v1.9.1+hotfix.2` works fine. Dependency summary **after the change (works)**:
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel unknown, v1.9.1+hotfix.2, on Mac OS X 10.14.4 18E226, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
[✓] Android Studio (version 3.3)
[✓] VS Code (version 1.38.1)
[✓] Connected device (1 available)
```

Prior dependency summary **(does not work):**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, v1.9.1+hotfix.4, on Mac OS X 10.14.4 18E226, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
[✓] Android Studio (version 3.3)
[✓] VS Code (version 1.38.1)
[✓] Connected device (1 available)
```

**Did something change** with `libimobiledevice` dependencies? 
* Previously instructions explicitly instructed how to install these dependencies, see June 30th page (Section: `Deploy to iOS devices`)  at https://web.archive.org/web/20190630180348/https://flutter.dev/docs/get-started/install/macos
* However, current instructions just ask to install CocoaPods.

**Comment by jmagman** [comment]
`libimobiledevice` is now packaged with `flutter` and does not need to be separately installed.  Do you have reason to think your issue is related to libimobiledevice?

**Comment by ketanm** [comment]
@jmagman  - No, I am not sure if there is any relation but I noticed this difference. 

I have not upgraded again but can do it if needed.

**Comment by rajeshzmoke** [comment]
```
Install failed

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      InstallCommand.runCommand (package:flutter_tools/src/commands/install.dart:44:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:490:18)
#3      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
#4      _rootRunUnary (dart:async/zone.dart:1132:38)
#5      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#6      _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
#7      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
#8      Future._propagateToListeners (dart:async/future_impl.dart:707:32)
#9      Future._completeWithValue (dart:async/future_impl.dart:522:5)
#10     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:552:7)
#11     _rootRun (dart:async/zone.dart:1124:13)
#12     _CustomZone.run (dart:async/zone.dart:1021:19)
#13     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#14     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#15     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#17     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:116:13)
#18     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:173:5)
```
getting the same error when doing `flutter install`

```
flutter doctor -v
[✓] Flutter (Channel stable, v1.9.1+hotfix.4, on Mac OS X 10.14.5 18F132, locale en-IN)
    • Flutter version 1.9.1+hotfix.4 at /Users/rajeshkumar/Desktop/flutter
    • Framework revision cc949a8e8b (2 weeks ago), 2019-09-27 15:04:59 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/rajeshkumar/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 28.0.3
    • ANDROID_HOME = /Users/rajeshkumar/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.0, Build version 11A420a
    • CocoaPods version 1.7.4

[✓] Android Studio (version 3.4)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 37.1.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[✓] IntelliJ IDEA Community Edition (version 2017.3.5)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 29.0.1
    • Dart plugin version 173.4700

[✓] VS Code (version 1.39.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.5.1

[✓] Connected device (1 available)
    • Hpsw iphone 6s • 23782bacce300171e42f2124196e5207f7b426a9 • ios • iOS 12.4.1

• No issues found!
```

**Comment by ketanm** [comment]
@rajeshzmoke, can you revert to prior version and see if it works? (`flutter version v1.9.1+hotfix.2`)

**Comment by tottakai** [comment]
Having similar issue with `flutter install`. If i run `flutter install --verbose`  i get: 

```
[  +13 ms] Installing com.xx.xx to iPhoneX...
[   +3 ms] executing: /Users/xxx/flutter/bin/cache/artifacts/ideviceinstaller/ideviceinstaller --list-apps
[   +6 ms] dyld: Library not loaded: /b/s/w/ir/k/homebrew/opt/libzip-flutter/lib/libzip.5.dylib
             Referenced from: /Users/xxx/flutter/bin/cache/artifacts/ideviceinstaller/ideviceinstaller
             Reason: image not found
[   +4 ms] executing: /Users/xxx/flutter/bin/cache/artifacts/ideviceinstaller/ideviceinstaller -i build/ios/iphoneos/Runner.app
[   +4 ms] dyld: Library not loaded: /b/s/w/ir/k/homebrew/opt/libzip-flutter/lib/libzip.5.dylib
             Referenced from: /Users/xxx/flutter/bin/cache/artifacts/ideviceinstaller/ideviceinstaller
             Reason: image not found
[  +11 ms] "flutter install" took 735ms.
Install failed

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      InstallCommand.runCommand (package:flutter_tools/src/commands/install.dart:44:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:490:18)
<asynchronous suspension>
#3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:407:33)
<asynchronous suspension>
#4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:29)
<asynchronous suspension>
#5      _rootRun (dart:async/zone.dart:1124:13)
#6      _CustomZone.run (dart:async/zone.dart:1021:19)
#7      _runZoned (dart:async/zone.dart:1516:10)
#8      runZoned (dart:async/zone.dart:1463:12)
#9      AppContext.run (package:flutter_tools/src/base/context.dart:153:18)
<asynchronous suspension>
#10     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:397:20)
#11     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
<asynchronous suspension>
#12     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:402:21)
<asynchronous suspension>
#13     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:29)
<asynchronous suspension>
#14     _rootRun (dart:async/zone.dart:1124:13)
#15     _CustomZone.run (dart:async/zone.dart:1021:19)
#16     _runZoned (dart:async/zone.dart:1516:10)
#17     runZoned (dart:async/zone.dart:1463:12)
#18     AppContext.run (package:flutter_tools/src/base/context.dart:153:18)
<asynchronous suspension>
#19     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:356:19)
<asynchronous suspension>
#20     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#21     new Future.sync (dart:async/future.dart:224:31)
#22     CommandRunner.run (package:args/command_runner.dart:112:14)
#23     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:242:18)
#24     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:22)
<asynchronous suspension>
#25     _rootRun (dart:async/zone.dart:1124:13)
#26     _CustomZone.run (dart:async/zone.dart:1021:19)
#27     _runZoned (dart:async/zone.dart:1516:10)
#28     runZoned (dart:async/zone.dart:1500:12)
#29     run.<anonymous closure> (package:flutter_tools/runner.dart:61:18)
<asynchronous suspension>
#30     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:154:29)
<asynchronous suspension>
#31     _rootRun (dart:async/zone.dart:1124:13)
#32     _CustomZone.run (dart:async/zone.dart:1021:19)
#33     _runZoned (dart:async/zone.dart:1516:10)
#34     runZoned (dart:async/zone.dart:1463:12)
#35     AppContext.run (package:flutter_tools/src/base/context.dart:153:18)
<asynchronous suspension>
#36     runInContext (package:flutter_tools/src/context_runner.dart:59:24)
<asynchronous suspension>
#37     run (package:flutter_tools/runner.dart:50:10)
#38     main (package:flutter_tools/executable.dart:65:9)
<asynchronous suspension>
#39     main (file:///Users/tokoskin/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#40     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:303:32)
#41     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
```

**Comment by zanderso** [comment]
It would also be helpful to have verbose logs from @ketanm and @rajeshzmoke with `flutter install --verbose`.

**Comment by ketanm** [comment]
@zanderso, my original post output is most likely from verbose. I will try again and post in a day.

**Comment by ketanm** [comment]
I have updated my previous comment. This looks like an issue related to `ideviceinstaller`

**Comment by christopherfujino** [comment]
So the `ideviceinstaller` issue is https://github.com/flutter/flutter/issues/42877, and I have a fix pending for this, which I hope to land to master tomorrow.

**Comment by christopherfujino** [comment]
This was fixed in #42877, which is available on the dev channel.
